addons/sale/models/sale_order.py

```diff
@@ -1271,6 +1271,31 @@ def _create_invoices(self, grouped=False, final=False, date=None):
         if final:
             moves.sudo().filtered(lambda m: m.amount_total < 0).action_switch_move_type()
         for move in moves:
+            if final and (dp_lines := move.line_ids.filtered('is_downpayment')):
+                # Downpayment might have been determined by a fixed amount set by the user.
+                # This amount is tax included. This can lead to rounding issues.
+                # E.g. a user wants a 100€ DP on a product with 21% tax.
+                # 100 / 1.21 = 82.64, 82.64 * 1,21 = 99.99
+                # This is already corrected by adding/removing the missing cents on the DP invoice,
+                # but must also be accounted for on the final invoice.
+                so_dp_invoices_total = sum(self.order_line.filtered('is_downpayment')\
+                    .invoice_lines.filtered(lambda x: x.move_id != move).mapped('price_total'))
+                delta_amount = so_dp_invoices_total + (sum(dp_lines.mapped('price_total')) * (1 if move.is_inbound() else -1))
+                if not self.currency_id.is_zero(delta_amount):
+                    product_lines = move.line_ids\
+                        .filtered(lambda aml: aml.display_type == 'product' and aml.is_downpayment)
+                    tax_lines = move.line_ids\
+                        .filtered(lambda aml: aml.tax_line_id.amount_type not in (False, 'fixed'))
+                    if tax_lines and product_lines:
+                        for lines, attr, sign in (
+                            (product_lines, 'price_total', -1),
+                            (tax_lines, 'amount_currency', 1),
+                        ):
+                            remaining = delta_amount
+                            for i, line in list(enumerate(lines))[::-1]:
+                                amt = remaining / (i+1)
```

Comment:
Maybe I'm missing something but I don't get the need for the `enumerate`, with n taxes on the sale order, we get `remaing/n`, `remaining / n-1`, ... 
is it not enough to calculate the amt once as
`amt = delta_amount / len(lines)` ?